src/libraries/Microsoft.Extensions.Logging.Console/tests/ConsoleLoggerExtensionsTests.cs

```diff
@@ -0,0 +1,357 @@
+                        {
+                            Indented = false,
+                            Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping
+                        };
+                    })
+                )
+                .BuildServiceProvider()
+                .GetRequiredService<ILoggerProvider>();
+
+            var consoleLoggerProvider = Assert.IsType<ConsoleLoggerProvider>(loggerProvider);
+            var logger = (ConsoleLogger)consoleLoggerProvider.CreateLogger("Category");
+            Assert.Equal(ConsoleFormatterNames.Json, logger.Options.FormatterName);
+            var formatter = Assert.IsType<JsonConsoleFormatter>(logger.Formatter);
+            Assert.Equal("HH:mm ", formatter.FormatterOptions.TimestampFormat);
+            Assert.True(formatter.FormatterOptions.UseUtcTimestamp);
+            Assert.True(formatter.FormatterOptions.IncludeScopes);
+            Assert.False(formatter.FormatterOptions.JsonWriterOptions.Indented);
+            Assert.Equal(JavaScriptEncoder.UnsafeRelaxedJsonEscaping, formatter.FormatterOptions.JsonWriterOptions.Encoder);
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/38337", TestPlatforms.Browser)]
+        public void AddConsole_NullFormatterNameUsingSystemdFormat_AnyDeprecatedPropertiesOverwriteFormatterOptions()
+        {
+            var configs = new[] {
+                new KeyValuePair<string, string>("Console:Format", "Systemd"),
+                new KeyValuePair<string, string>("Console:TimestampFormat", "HH:mm:ss "),
+                new KeyValuePair<string, string>("Console:FormatterOptions:TimestampFormat", "HH:mm "),
+            };
+            var configuration = new ConfigurationBuilder().AddInMemoryCollection(configs).Build();
+
+            var loggerProvider = new ServiceCollection()
+                .AddLogging(builder => builder
+                    .AddConfiguration(configuration)
+                    .AddConsole(o => { })
+                )
+                .BuildServiceProvider()
+                .GetRequiredService<ILoggerProvider>();
+
+            var consoleLoggerProvider = Assert.IsType<ConsoleLoggerProvider>(loggerProvider);
+            var logger = (ConsoleLogger)consoleLoggerProvider.CreateLogger("Category");
+            Assert.Null(logger.Options.FormatterName);
+            var formatter = Assert.IsType<SystemdConsoleFormatter>(logger.Formatter);
+            Assert.Equal("HH:mm:ss ", formatter.FormatterOptions.TimestampFormat);
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/38337", TestPlatforms.Browser)]
+        public void AddConsole_NullFormatterNameUsingDefaultFormat_AnyDeprecatedPropertiesOverwriteFormatterOptions()
```

Comment:
This is the current behavior with regards to what is getting picked for the properties.

I chatted with @eerhardt offline and he recommends we should not be picking any properties from under "Console:FormatterOptions" if FormatterName is not specified.

Here are the expectations I need to satisfy:
- [ ] when formatter name available skip any deprecated APIs (instead as you see in this test when FormatterName is missing, then deprecated Console properties overwrite new ones under Console:FormatterOptions)
- [x] when formatter name not provided fall back to using deprecated Format and properties and ignore any formatter options
- [x] when formatter name is not found fall back to simple